From edb2479714b770e8cffd1b56e5e6a6e37d1ded59 Mon Sep 17 00:00:00 2001
From: sajcho <saux.aarch64@gmail.com>
Date: Wed, 16 Jun 2021 17:43:29 +0200
Subject: [PATCH] pkgadd.conf: runit rules

---
 pkgadd.conf | 7 +++----
 1 file changed, 3 insertions(+), 4 deletions(-)

diff --git a/pkgadd.conf b/pkgadd.conf
index 4ac0cb9..89a5a1f 100644
--- a/pkgadd.conf
+++ b/pkgadd.conf
@@ -13,12 +13,11 @@ UPGRADE		^var/run/utmp$		NO
 UPGRADE		^etc/ports/drivers/.*$	YES
 UPGRADE		^etc/X11/.*$		YES
 
-UPGRADE		^etc/rc.*$		YES
+UPGRADE		^etc/runit/.*$		NO
+UPGRADE		^etc/sv/.*$		NO
 UPGRADE		^etc/rc\.local$		NO
-UPGRADE		^etc/rc\.modules$	NO
+UPGRADE		^etc/rc\.shutdown$	NO
 UPGRADE		^etc/rc\.conf$		NO
-UPGRADE		^etc/rc\.d/net$		NO
-UPGRADE		^etc/rc\.d/wlan$	NO
 
 UPGRADE		^etc/udev/rules.d/.*$	YES
 UPGRADE		^etc/udev/rules.d/1.*$	NO
-- 
2.31.1

From 9368f95f9452e31e81d0be530315047f83268369 Mon Sep 17 00:00:00 2001
From: sajcho <saux.aarch64@gmail.com>
Date: Wed, 16 Jun 2021 17:46:52 +0200
Subject: [PATCH] pkgmk.conf: define multiarch variable

---
 pkgmk.conf | 23 ++++++++++++-----------
 1 file changed, 12 insertions(+), 11 deletions(-)

diff --git a/pkgmk.conf b/pkgmk.conf
index 781e8b9..32bdf6b 100644
--- a/pkgmk.conf
+++ b/pkgmk.conf
@@ -2,20 +2,17 @@
 # /etc/pkgmk.conf: pkgmk(8) configuration
 #
 
-export CFLAGS="-O2 -march=x86-64 -pipe"
-export CXXFLAGS="${CFLAGS}"
-
-# export JOBS=$(nproc)
-# export MAKEFLAGS="-j $JOBS"
+export CFLAGS="-O2 -pipe"
+export CXXFLAGS="-O2 -pipe"
 
 case ${PKGMK_ARCH} in
-	"64"|"")
+	"x86_64")
+		export CFLAGS="${CFLAGS} -march=x86-64"
+		export CXXFLAGS="${CXXFLAGS}"
 		;;
-	"32")
-		export CFLAGS="${CFLAGS} -m32"
-		export CXXFLAGS="${CXXFLAGS} -m32"
-		export LDFLAGS="${LDFLAGS} -m32"
-		export PKG_CONFIG_LIBDIR="/usr/lib32/pkgconfig"
+	"aarch64")
+		export CFLAGS="${CFLAGS} -march=armv8-a"
+		export CXXFLAGS="${CXXFLAGS}"
 		;;
 	*)
 		echo "Unknown architecture selected! Exiting."
@@ -23,6 +20,10 @@ case ${PKGMK_ARCH} in
 		;;
 esac
 
+# export JOBS=$(nproc)
+# export MAKEFLAGS="-j $JOBS"
+# export SAMUFLAGS="-j $JOBS"
+
 # PKGMK_SOURCE_MIRRORS=()
 # PKGMK_SOURCE_DIR="$PWD"
 # PKGMK_PACKAGE_DIR="$PWD"
-- 
2.31.1

From 61e366de06a4b7bdd7827419c24cb4a43370654a Mon Sep 17 00:00:00 2001
From: sajcho <saux.aarch64@gmail.com>
Date: Wed, 16 Jun 2021 17:50:21 +0200
Subject: [PATCH] pkgutil.cc: add zstd support

---
 pkgutil.cc | 1 +
 1 file changed, 1 insertion(+)

diff --git a/pkgutil.cc b/pkgutil.cc
index e72d817..b97e451 100644
--- a/pkgutil.cc
+++ b/pkgutil.cc
@@ -48,6 +48,7 @@
 	archive_read_support_filter_bzip2((ar)); \
 	archive_read_support_filter_xz((ar)); \
 	archive_read_support_filter_lzip((ar)); \
+	archive_read_support_filter_zstd((ar)); \
 	archive_read_support_format_tar((ar))
 
 #define DEFAULT_BYTES_PER_BLOCK (20 * 512)
-- 
2.31.1

From 1240435fc0dfe1c67d64f9d949d192bf5174ce25 Mon Sep 17 00:00:00 2001
From: sajcho <saux.aarch64@gmail.com>
Date: Wed, 16 Jun 2021 18:04:03 +0200
Subject: [PATCH] pkgmk.in: add support zstd and multiarch

---
 pkgmk.in | 16 +++++++++-------
 1 file changed, 9 insertions(+), 7 deletions(-)

diff --git a/pkgmk.in b/pkgmk.in
index 7b88fb1..08b7f35 100755
--- a/pkgmk.in
+++ b/pkgmk.in
@@ -197,7 +197,7 @@ unpack_source() {
 	for FILE in ${source[@]}; do
 		LOCAL_FILENAME=`get_filename $FILE`
 		case $LOCAL_FILENAME in
-			*.tar|*.tar.gz|*.tar.Z|*.tgz|*.tar.bz2|*.tbz2|*.tar.xz|*.txz|*.tar.lzma|*.tar.lz|*.zip|*.rpm|*.7z)
+			*.tar|*.tar.gz|*.tar.Z|*.tgz|*.tar.bz2|*.tbz2|*.tar.xz|*.txz|*.tar.lzma|*.tar.lz|*.zip|*.rpm|*.7z|*.zst)
 				COMMAND="bsdtar -p -o -C $SRC -xf $LOCAL_FILENAME" ;;
 			*)
 				COMMAND="cp $LOCAL_FILENAME $SRC" ;;
@@ -553,6 +553,7 @@ build_package() {
 			bz2) COMPRESSION="-j" ;;
 			xz)  COMPRESSION="-J" ;;
 			lz)  COMPRESSION="--lzip" ;;
+			zst) COMPRESSION="--zstd" ;;
 		esac
 		bsdtar -c $COMPRESSION -f $TARGET *  &&  bsdtar -t -v -f $TARGET
 
@@ -792,10 +793,11 @@ parse_options() {
 }
 
 main() {
-	# multilib
-	PKGMK_ARCH=64
-	if [ -f ".32bit" ]; then
-		PKGMK_ARCH=32
+	# multiarch
+	if [ `uname -m` = "x86_64" ]; then
+		PKGMK_ARCH="x86_64"
+	elif [ `uname -m` = "aarch64" ]; then
+		PKGMK_ARCH="aarch64"
 	fi
 
 	local FILE TARGET
@@ -822,8 +824,8 @@ main() {
 	check_pkgfile
 
 	case $PKGMK_COMPRESSION_MODE in
-		gz|bz2|xz|lz)
-			TARGET="$PKGMK_PACKAGE_DIR/$name#$version-$release.pkg.tar.$PKGMK_COMPRESSION_MODE"
+		gz|bz2|xz|lz|zst)
+			TARGET="$PKGMK_PACKAGE_DIR/$name#$version-$release.$PKGMK_ARCH.pkg.tar.$PKGMK_COMPRESSION_MODE"
 			;;
 		*)
 			error "Compression mode '$PKGMK_COMPRESSION_MODE' not supported"
-- 
2.31.1

